presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1198,6 +1200,63 @@ velox::core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
       ROW(std::move(outputNames), std::move(outputTypes)));
 }
 
+
+std::shared_ptr<const velox::core::IndexLookupJoinNode>
+VeloxQueryPlanConverterBase::toVeloxQueryPlan(
+  const std::shared_ptr<const protocol::IndexJoinNode>& node,
+  const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+  const protocol::TaskId& taskId) {
+    std::vector<core::FieldAccessTypedExprPtr> leftKeys;
+    std::vector<core::FieldAccessTypedExprPtr> rightKeys;
+
+    leftKeys.reserve(node->criteria.size());
+    rightKeys.reserve(node->criteria.size());
+    for (const auto& clause : node->criteria) {
+      leftKeys.emplace_back(exprConverter_.toVeloxExpr(clause.left));
+      rightKeys.emplace_back(exprConverter_.toVeloxExpr(clause.right));
+    }
+
+    auto left = toVeloxQueryPlan(node->probeSource, tableWriteInfo, taskId);
+    auto right = toVeloxQueryPlan(node->indexSource, tableWriteInfo, taskId);
+
+    std::vector<std::string> outputNames = left->outputType()->names();
+    std::vector<std::string> rightNames = right->outputType()->names();
+    outputNames.insert(outputNames.end(), rightNames.begin(), rightNames.end());
+    std::vector<TypePtr> outputTypes = left->outputType()->children();
+    std::vector<TypePtr> rightTypes = right->outputType()->children();
+    outputTypes.insert(outputTypes.end(), rightTypes.begin(), rightTypes.end());
+
+    const std::vector<velox::core::IndexLookupConditionPtr> joinConditions;
+
+    return std::make_shared<core::IndexLookupJoinNode>(
+        node->id,
+        core::JoinType::kInner,
+        /*leftKeys=*/leftKeys,
+        /*rightKeys=*/rightKeys,
+        /*joinConditions=*/joinConditions /*std::vector<IndexLookupConditionPtr>*/,
+        /*left=*/left /*PlanNodePtr*/,
+        /*right=*/std::dynamic_pointer_cast<const core::TableScanNode>(right) /*TableScanNodePtr*/,
+        /*outputType=*/ROW(std::move(outputNames), std::move(outputTypes)));
```

Comment:
IndexJoinNode shall provide outputVariables as const protocol::JoinNode does?

Reply:
IndexJoinNode in Java doesn't provide outputVariables